File: 0053-maximum-subarray/0053-maximum-subarray.java
class Solution {
    public int maxSubArray(int[] nums) {
        if(nums.length < 1) return 0;
        if (nums.length == 1) return nums[0];
        int maxSum = nums[0];
        int Sum = nums[0];
        for (int i = 1; i < nums.length; i ++){
            Sum = Math.max(Sum+=nums[i], nums[i]);
            maxSum = Math.max(Sum, maxSum);
        }
        return maxSum;
    }
}

File: 0053-maximum-subarray/0053-maximum-subarray.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {

        int curr_sum =  0;
        int max_number = INT_MIN;

        for (int i = 0 ; i< nums.size() ; i++ ) {

       
        curr_sum =  max(nums[i] , nums[i] + curr_sum);
        max_number = max(max_number , curr_sum);
        }
         return max_number;
        
    }
   
};